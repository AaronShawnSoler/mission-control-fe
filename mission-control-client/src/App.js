import React, { useEffect, useState } from "react";
import {
  Switch,
  Route,
  useLocation,
  Redirect
} from "react-router-dom";
import { Security, ImplicitCallback } from "@okta/okta-react";
// components
import Layout from "./components/layout/Layout";
// Context
import {ProductContext} from './context/ProductContext'
import ProductContextProvider from './context/ProductContextProvider';
// AXIOS
import axiosLabsGraphQL from "./utils/axiosLabsGraphQL";
import axios from "axios";
// GRAPHQL
import { fullProjectDetailsById, peopleByProjectId } from "./queries";

// import Registration from "./components/auth/Registration.js";
import Login from "./components/auth/Login";
import ProjectMore from "./components/dashboard/admin-dashboard/ProjectMore";
import Bad from "./components/layout/Bad";
import AdminDash from "./components/dashboard/admin-dashboard/DashboardHome";
import UserDash from "./components/dashboard/user-dashboard/DashboardHome";

//utils
// import AdminPrivateRoute from "./utils/AdminPrivateRoute";
// import PrivateRoute from "./utils/PrivateRoute";
// import StudentPrivateRoute from "./utils/StudentPrivateRoute";
import embedAnalytics from "./utils/embedAnalytics";

//styles
import "./styles/index.scss";

function App() {
  
    // State for context
    // const [productState, setProductState] = React.useState(
    //   {
    //     isLoading: false,
    //     err: null,
    //     active: null,
    //     project: null
    //   }
    // )
  
    // // Product Context fn's
    // const setActiveProduct = el => {
    //   setProductState({...productState, active: el})
    // }
  
    // const setSelectedProject = id => {
    //   axios
    //       .all([
    //         axiosLabsGraphQL.post("", { query: fullProjectDetailsById(id) }),
    //         axiosLabsGraphQL.post("", { query: peopleByProjectId(id) })
    //       ])
    //       .then(
    //         axios.spread((res, res2) => {
    //           const project = {
    //             project: res.data.data.projects,
    //             people: res2.data.data.projectRoles
    //           }
    //           // dispatch({ type: SET_ACTIVE_PROJECT_SUCCESS, payload: project });
    //           setProductState({...productState, project: project})
    //         })
    //       )
    //       .catch(err => {
    //         // dispatch({ type: SET_ACTIVE_PROJECT_FAILURE, payload: err.response });
    //         console.log(err)
    //       });
    // }


  const location = useLocation();

  useEffect(() => {
    embedAnalytics();
  }, [location]);

  function onAuthRequired({ history }) {
    history.push("/");
  }

  return (
    // <ProductContext.Provider value={{productState, setActiveProduct, setSelectedProject}}>
      <Layout>
        <Security
          issuer={`${process.env.REACT_APP_OKTA_SERVER}`}
          clientId={`${process.env.REACT_APP_OKTA_CLIENT}`}
          redirectUri={window.location.origin + "/implicit/callback"}
          onAuthRequired={onAuthRequired}
          pkce={true}
        >
          <Switch>
            {/* {(!localStorage.getItem("role") ||
              !["admin", "manager", "student"].includes(decrypt())) &&
              localStorage.removeItem("token") &&
              history.push("/login")} */}
            {/* <PrivateRoute path="/" exact>
              {localStorage.getItem("token") ? (
                <Redirect to={{ pathname: `${decrypt()}/dashboard` }} />
              ) : (
                <Redirect to="/login" />
              )}
            </PrivateRoute> */}
            {/* OKTA will be taking over user access and control */}
            <Route
              path="/login"
              render={() => <Login baseUrl={`${process.env.REACT_APP_OKTA_URL}`} />}
            />
            <Route exact={true} path="/">
              {localStorage.getItem("okta-token-storage") ? 
              <Redirect push to="/admin/dashboard" />
            :  (
              <Redirect push to="/login" />
            )}
            </Route>
            <Route path="/student/dashboard" component={UserDash} />
            <Route exact path="/manager/dashboard" component={AdminDash} />
            <Route exact path="/admin/dashboard" component={AdminDash} />
            <ProductContextProvider>
              <Route path="/admin/dashboard/:id" component={ProjectMore} />
            </ProductContextProvider>
            {/* OKTA Signin Widget route */}
            <Route path="/implicit/callback" component={ImplicitCallback} />
            <Route component={Bad} />
          </Switch>
        </Security>
      </Layout>
    // </ProductContext.Provider>
  );
}

export default App;
