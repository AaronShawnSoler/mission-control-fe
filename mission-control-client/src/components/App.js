import React from 'react';
import { Route, Switch, useLocation } from 'react-router-dom';
import { SecureRoute, ImplicitCallback } from '@okta/okta-react';

import { embedAnalytics } from '../utils';

import Authorization from './Authorization';
import Project from './Project/Project';
import ProjectListView from './ProjectListView';

const App = () => {
  const location = useLocation();

  React.useEffect(() => {
    embedAnalytics();
  }, [location]);

  return (
    <Switch>
      <Route path="/implicit/callback" component={ImplicitCallback} />
      <Route path="/project" component={Project} />
      {/* Authorization component renders ProjectListView implicitly */}
      <Route path="/" component={ProjectListView} />
      <SecureRoute exact path="/" component={Authorization} />
    </Switch>
  );
};

export default App;
