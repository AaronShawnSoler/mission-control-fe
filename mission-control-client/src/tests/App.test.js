import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import App from '../App';
import * as rtl from '@testing-library/react';
afterEach(rtl.cleanup);

describe('App', () => {
    it ('renders without crashing', () => {
        const history = createMemoryHistory();
        const wrapper = rtl.render(
            <Router history={history}>
                <App />
            </Router>);
        expect(wrapper.baseElement).toMatchSnapshot();
    });
});
