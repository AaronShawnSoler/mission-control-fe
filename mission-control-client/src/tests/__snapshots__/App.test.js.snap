// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<body>
  <div>
    <div
      class="site-container"
    >
      <div
        class="nav-container"
      >
        <div
          class="nav-logo"
        >
          <a
            class="nav-head"
            href="/login"
          >
            MISSION CONTROL
          </a>
        </div>
        <div
          class="nav-links"
        >
          <a
            aria-current="page"
            class="nav-link selected"
            data-testid="signin"
            href="/login"
          >
            Sign In
          </a>
          <a
            class="nav-link signup"
            data-testid="signup"
            href="/register"
          >
            Sign Up
          </a>
        </div>
      </div>
      <div
        style="position: relative;"
      >
        <div
          class="auth-container"
        >
          <h1
            class="auth-header"
            data-testid="signin-head"
          >
            Sign in
          </h1>
          <p
            class="dontHave"
          >
            Don't have an account? 
            <a
              href="/register"
            >
              Create One
            </a>
          </p>
          <form
            class="login-form"
            data-testid="login-form"
            history="[object Object]"
          >
            <div
              class="email"
            >
              <label
                for="email"
              >
                Email
              </label>
              <input
                data-testid="email-field"
                name="email"
                placeholder="Enter Your Email. . ."
                type="text"
                value=""
              />
            </div>
            <div
              class="password"
            >
              <label
                for="password"
              >
                Password
              </label>
              <input
                data-testid="password-field"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
            <div
              class="remember"
            >
              <input
                class="checkbox"
                name="remembered"
                type="checkbox"
                value=""
              />
              <p>
                Remember me
              </p>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text btn MuiButton-textPrimary"
              data-testid="submit"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                LOG IN
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
        </div>
        <img
          alt="group of people working on their laptops"
          class="auth-img"
          src="computers.svg"
        />
      </div>
    </div>
  </div>
</body>
`;
